libcontainerd/client_linux.go

```diff
@@ -445,12 +447,43 @@ func (clnt *client) restore(cont *containerd.Container, options ...CreateOption)
 }
 
 func (clnt *client) Restore(containerID string, options ...CreateOption) error {
+	if clnt.liveRestore {
```

Comment:
I'm confused why we kill all container processes if they are still hanging around when we start? I understand that we need `--live-restore` in order to maintain back-compat when the daemon dies. But it seems odd that we go from the old code (which did "live restore" by default) to the new code that will **kill** all of the containers if they're still lying around. Does this play nicely when you have more than one consumer of `containerd`?